crates/fuel-streams-macros/src/lib.rs

```diff
@@ -5,17 +5,18 @@ pub mod subject {
 
     /// This trait is used internally by the `Subject` derive macro to convert a struct into a
     /// standard NATS subject.
-    pub trait IntoSubject: std::fmt::Debug + Clone + Default {
-        const WILDCARD: &'static str;
-
-        fn parse_param<V: ToString>(param: &Option<V>) -> String {
-            match param {
-                Some(val) => val.to_string(),
-                None => "*".to_string(),
-            }
-        }
-
+    pub trait IntoSubject: std::fmt::Debug + Send + Sync {
         fn parse(&self) -> String;
+    }
+
+    pub trait SubjectBuildable: std::fmt::Debug {
```

Comment:
agree

Reply:
> Why do we need another trait here? Isn't it less overhead for our users to import just one trait instead of 2?

https://github.com/FuelLabs/data-systems/blob/be6ff0726e5d17f6bcf7de3879095d1a60b1515a/crates/fuel-streams/src/stream/stream_impl.rs#L19-L20

Yeah, I agree too. It would be really nice to avoid the extra trait. However, the `Filter` interface seems to use `new()`, which isn't object-safe for `IntoSubject`. Hence, the need for the extra trait, `SubjectBuildable`

Reply:
fine by me.